src/agent/onefuzz-agent/src/tasks/analysis/single.rs

```diff
@@ -0,0 +1,124 @@
+        set_executable(&tools.local_path)
+            .await
+            .context("to set tools as executable")?;
+    }
+
+    run_tool(&config).await
+}
+
+pub async fn run_tool(config: &Config) -> Result<()> {
+    let heartbeat = config.common.init_heartbeat(Some(INITIAL_DELAY)).await?;
+    let expand = Expand::new()
+        .target_exe(&config.target_exe)
+        .target_options(&config.target_options)
+        .analyzer_exe(&config.analyzer_exe)
+        .analyzer_options(&config.analyzer_options)
+        .crashes(&config.crashes.local_path)
+        .set_optional_ref(&config.tools, |tester, key| {
+            tester.tools_dir(&key.local_path)
+        })
+        .setup_dir(&config.common.setup_dir)
+        .job_id(&config.common.job_id)
+        .task_id(&config.common.task_id)
+        .set_optional_ref(&config.common.microsoft_telemetry_key, |tester, key| {
+            tester.microsoft_telemetry_key(&key)
+        })
+        .set_optional_ref(&config.common.instance_telemetry_key, |tester, key| {
+            tester.instance_telemetry_key(&key)
+        })
+        .set_optional_ref(
+            &config.crashes.remote_path.clone().and_then(|u| u.account()),
+            |tester, account| tester.crashes_account(account),
+        )
+        .set_optional_ref(
+            &config
+                .crashes
+                .remote_path
+                .clone()
+                .and_then(|u| u.container()),
+            |tester, container| tester.crashes_container(container),
+        );
+
+    let input_path = expand
+        .evaluate_value(format!("{{crashes}}/{}", config.input_file))
+        .context("unable to expand input_path")?;
+    let expand = expand.input_path(input_path);
+
+    let analyzer_path = expand
+        .evaluate_value(&config.analyzer_exe)
+        .context("expanding analyzer_exe failed")?;
+
+    loop {
+        let mut cmd = Command::new(&analyzer_path);
+        cmd.kill_on_drop(true)
+            .env_remove("RUST_LOG")
+            .stdin(Stdio::null())
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped());
+
+        for arg in expand.evaluate(&config.analyzer_options)? {
+            cmd.arg(arg);
+        }
+
+        for (k, v) in &config.analyzer_env {
+            cmd.env(
+                k,
+                expand
+                    .evaluate_value(v)
+                    .context("expanding analyzer_env failed")?,
+            );
+        }
+
+        info!("analyzing input with {:?}", cmd);
+        let output = cmd
+            .spawn()
+            .with_context(|| format!("analyzer failed to start: {}", analyzer_path))?;
+
+        heartbeat.alive();
+
+        // while we monitor the runtime of the debugger, we don't fail the task if
+        // the debugger exits non-zero. This frequently happens during normal use of
+        // debuggers.
+        monitor_process(output, "crash-repro".to_string(), true, None)
```

Comment:
shouldn't we be storing the logs of the analysis ?

Reply:
Yes.  `monitor_process` logs stdout and stderr to application insights.